server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1389,12 +1418,39 @@ public static Setting<Integer> intSetting(
         return new Setting<>(
             key,
             Integer.toString(defaultValue),
-            (s) -> parseInt(s, minValue, maxValue, key, isFiltered(properties)),
+            new IntegerParser(key, minValue, maxValue, isFiltered(properties)),
             validator,
             properties
         );
     }
 
+    private static class IntegerParser implements Function<String, Integer>, Writeable {
+        private String key;
+        private int minValue;
+        private int maxValue;
+        private boolean isFiltered;
+
+        public IntegerParser(String key, int minValue, int maxValue, boolean isFiltered) {
```

Comment:
Same as earlier comment on matching the same parameter order as the `parseInt` method.